sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/ResolvedDataSource.scala

```diff
@@ -228,7 +228,7 @@ object ResolvedDataSource extends Logging {
           sqlContext,
           fileCatalog,
           partitionSchema = partitionSchema,
-          dataSchema = dataSchema,
+          dataSchema = dataSchema.asNullable,
```

Comment:
This is to keep consistent with the previous behaviour, i.e. data schema is always nullable, see https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/ResolvedDataSource.scala#L170-L171